gms/gossiper.cc

```diff
@@ -1425,14 +1420,16 @@ endpoint_state_ptr gossiper::get_endpoint_state_ptr(inet_address ep) const noexc
     }
 }
 
-future<> gossiper::update_address_map(const rpc::client_info& cinfo) noexcept {
+future<endpoint_id> gossiper::get_endpoint_id(const rpc::client_info& cinfo) noexcept {
     auto host_id = cinfo.retrieve_auxiliary<locator::host_id>("host_id");
     auto addr = cinfo.retrieve_auxiliary<gms::inet_address>("baddr");
 
     if (host_id) {
-        return update_address_map(host_id, addr);
+        co_await update_address_map(host_id, addr);
+    } else {
+        host_id = get_host_id(addr, throw_on_error::no);
```

Comment:
I'm not sure we can trust nodes running old versions to always have the HOST_ID in their application state.

Reply:
It's set together with everything else in prepare_to_join(), and the line dates back to 2015:
```
    app_states.emplace(gms::application_state::HOST_ID, value_factory.host_id(local_host_id));
```
So it's pretty much there since the beginning.

Reply:
Maybe I needed it due to a bug I had in the series with updating the endpoint_state.
I'll check if we can mandate a valid host_id here.